### PR TITLE
Apids (Bee People) Approved™ Pectoral Edition

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

This PR introduces a new playable humanoid race, Apids, bee people in other words.

image

## Why It's Good For The Game

It gives the player more options to play as, and well, I mean, we're on BeeStation. This'll probably get replaced with a more in-depth reason as soon as I can get the opinions of some folk regarding certain ideas. It also replaces flypeople, who while cool, didn't really have anything majorly game-changing about them, as a selectable race.

## Changelog
:cl:
add: Added apids, also known as bee people.
add: Added a whole lot of shit related to apids.
/:cl:

## Done, but will probably need fixing-up and such

- Existence
- Take more damage to flyswatters
- Take more damage to pestkillers
- Can breathe in pressures of 8kPA thanks to the many spiracles in their body (bullshit).
- The Apidite language, consisting of one syllable: Bzz.
- Apid Wings that permanently take up your neck slot and allow you to dash forward 4 tiles.
- Take twice as much heat, cold, burn, and stamina damage.
- Roundstart enablement with disablement of flypeople.

## Future

- Adds the apid stinger - A stinger that upon stinging, injects 5u of Toxins every second it remains embedded. If the stung person moves, the stinger and liver of the apid gets removed. If the apid moves, the apid takes 30 brute damage and the stung person 20. Only careful yanking-out works as the non-harmful way to remove it.
- Get high off smoke.

## Possible Ideas
- Apid comradeship with bees, aka summoning bees from REASONABLY nearby beehives to maybe heal you or help you bash in the skull of an assistant.
- Have hornet people be the ash-walkers of the jungle, if that ever happens.

## Credits

- thatguythere03 for spriting the bodyparts and suggestions
- Ayy-Robotics, Emotions, Dekunutkid, for suggestions in Discord.
- The rest of BeeStation's Discord, for suggestions and somewhat support.

### CODE REVIEW

**Comment by Crossedfall** [comment]
When you're ready for review convert this PR from a draft

**Comment by nikk735** [comment]
kinda sexy doe

**Comment by Oblivire** [comment]
For another possible idea, maybe using a container on them, targeting the groin (since there's no tail), on help intent could put 5u of toxins into the container?

**Comment by ghost** [comment]
This is ready for review, assuming it passes Travis. I'd like to note that the smoked effect and stingers will be added afterwards if Apids do get merged.

**Comment by ghost** [comment]
I also enabled maintainer editing, just in case.

**Comment by qwertyquerty** [comment]
Notify me when this is ready.

**Comment by ghost** [comment]
> Notify me when this is ready.

It's ready.

File: code/modules/mob/living/carbon/human/species_types/apid.dm [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	var/randname = apid_name(gender)
+
+	if(lastname)
+		randname += " [lastname]"
+
+	return randname
+
+/datum/species/apid/on_species_gain(mob/living/carbon/C, datum/species/old_species, pref_load)
+	. = ..()
+	C.equip_to_slot_or_del(new /obj/item/clothing/shoes/bhop/apid(C), SLOT_NECK)
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/human/species_types/apid.dm]: The growing wings should be forced (drop current neck item or something.)

Comment (ghost) [reply code/modules/mob/living/carbon/human/species_types/apid.dm]: Alright, that should fix it.

**Comment by qwertyquerty** [comment]
Checks really do be failing doe

**Comment by ghost** [comment]
Big sad fail moment ngl @qwertyquerty

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -13,7 +13,7 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 	var/list/offset_features = list(OFFSET_UNIFORM = list(0,0), OFFSET_ID = list(0,0), OFFSET_GLOVES = list(0,0), OFFSET_GLASSES = list(0,0), OFFSET_EARS = list(0,0), OFFSET_SHOES = list(0,0), OFFSET_S_STORE = list(0,0), OFFSET_FACEMASK = list(0,0), OFFSET_HEAD = list(0,0), OFFSET_FACE = list(0,0), OFFSET_BELT = list(0,0), OFFSET_BACK = list(0,0), OFFSET_SUIT = list(0,0), OFFSET_NECK = list(0,0))
 
 	var/hair_color	// this allows races to have specific hair colors... if null, it uses the H's hair/facial hair colors. if "mutcolor", it uses the H's mutant_color
-	var/hair_alpha = 255	// the alpha used by the hair. 255 is completely solid, 0 is transparent.
+	var/hair_alpha = 255	// the alpha used by the hair. 255 is cmpletely solid, 0 is transparent.
```
Comment (qwertyquerty) [inline code/modules/mob/living/carbon/human/species.dm]: You added a random typo

**Comment by qwertyquerty** [comment]
After reviewing all of this, I feel like there just isn't enough that makes apids unique yet

**Comment by Zenithrium** [comment]
maybe apids could make honey in some way?

**Comment by PowerfulBacon** [comment]
> 
> 
> After reviewing all of this, I feel like there just isn't enough that makes apids unique yet

They are more unique than fly-people however and would be better as a roundstart role than them

**Comment by qwertyquerty** [comment]
Will merge after the random typo is fixed. I expect the mentioned follow-up changes

File: code/modules/mob/living/carbon/human/species.dm (hunk 2 above)
Comment (ghost) [reply code/modules/mob/living/carbon/human/species.dm]: Huh, okay.

**Pull request merged**